### ISSUE TITLE
Issue with grdview @relief_30s -G@day_01m

### ISSUE THREAD

**Issue by Esteban82** [GenericMappingTools/gmt#5959]
I try the following. I use a relief grid of 30 s and a drape file of minor resolution (01m). Is it a bug? It this suppose to work?

`gmt grdview @earth_relief_30s -RBO -JM14c -JZ2c -p160/30 -Qi300 -N0+glightgray -BnSwEZ -Baf -Bzaf+l"Altura (m)" -G@earth_day_01m -png prueba
`
I got this error:

```

grdview (gmtapi_init_grdheader): Please select compatible -R and -I values
free(): invalid next size (normal)
Abortado (`core' generado)

```

With the same resolution, it works:

`gmt grdview @earth_relief_01m -RBO -JM14c -JZ2c -p160/30 -Qi300 -N0+glightgray -BnSwEZ -Baf -Bzaf+l"Altura (m)" -G@earth_day_01m -png prueba`

![prueba](https://user-images.githubusercontent.com/17908951/140676321-366efce3-6d39-4280-913d-821f25a3da3e.png)

**Comment by Esteban82** [comment]
I remove ` -N0+glightgray` and I get

```
gmt grdview @earth_relief_30s -RBO -JM14c -JZ2c -p160/30 -Qi300 -BnSwEZ -Baf -Bzaf+l"Altura (m)" -G@earth_day_01m -png prueba
grdview [WARNING]: (x_max-x_min) must equal (NX + eps) * x_inc), where NX is an integer and |eps| <= 0.0001.
grdview [WARNING]: (y_max-y_min) must equal (NY + eps) * y_inc), where NY is an integer and |eps| <= 0.0001.
grdview (gmtapi_init_grdheader): Please select compatible -R and -I values

```
![prueba](https://user-images.githubusercontent.com/17908951/140677740-19268f88-d131-4cad-8126-4492c8d6a600.png)

**Comment by meghanrjones** [comment]
Fixed by #5961
